File: problems/sqrtx/binary-search.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0
        if x == 1:
            return 1

        lower = 0
        upper = x

        while lower < upper:
            mid = (lower + upper) // 2
            result = mid * mid
            print('r  = ' + str(mid) + ' total = ' + str(result))
            if result == x:
                return mid

            if result > x:
                upper = mid
            else:
                lower = mid + 1

        return lower - 1

File: problems/guess-number-higher-or-lower/binary-search.cs
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

public class Solution : GuessGame {
    public int GuessNumber(int n) {
        var low = 1;
        var high = n;
        while (low <= high)
        {
            var mid = low + (high - low) / 2;
            var res = guess(mid);
            if (res == 0) return mid;
            if (res == -1)
            {
                high = mid - 1;
                continue;
            }

            low = mid + 1;
        }

        return -1;
    }
}